internal/terminal/log.go

```diff
@@ -0,0 +1,117 @@
+package terminal
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/iancoleman/orderedmap"
+)
+
+// LogLevel is the level of a terminal log
+type LogLevel string
+
+// set of supported log levels
+const (
+	LogLevelInfo  LogLevel = "info"
+	LogLevelError LogLevel = "error"
+)
+
+var (
+	allLogLevels = []LogLevel{LogLevelInfo, LogLevelError}
+
+	longestLogLevel = func() LogLevel {
+		var longest LogLevel
+		var longestLength int
+		for _, level := range allLogLevels {
+			if l := len(level); l > longestLength {
+				longest = level
+				longestLength = l
+			}
+		}
+		return longest
+	}()
+)
+
+// LogData produces the log data
+type LogData interface {
+	Message() (string, error)
+	Payload() ([]string, map[string]interface{}, error)
+}
+
+// Log is a terminal log
+type Log struct {
+	Level LogLevel
+	Time  time.Time
+	TZ    *time.Location
```

Comment:
⚠️ Setting a time zone here _works_, but doesn't feel great

The motivation behind it was to support displaying a `time.Local` formatted timestamp to the user when running `realm-cli` (hence the defaults in all `NewXXXLog` constructors), but allow for it to be overridden (specifically in tests, so that expected timestamps can be written in UTC)

Open to ideas here!

Reply:
hmm do we _need_ local timestamps/is this a requirement? can we just use UTC everywhere and make everyone's lives easier?

Reply:
no hard requirement here, I've just been trying to keep output as simple as possible, and this was a way to shave 5 characters of output off.

if no better way here for localizing a log, I'll revert back to UTC for everything (and/or remove the timestamp all together... I'll ask product/design about that one tomorrow at the checkin)

Reply:
I agree with Chris here if there's no hard requirement. Better to make it simpler for us and not have to deal with timezones

Reply:
after checking with product and design in the checkin, timezones are now no more.  this has been removed